accepted/2020/binary-data/binary-data.md

```diff
@@ -0,0 +1,177 @@
+queue. The template already created a handler `RunAsync` that passes the queue
+item as an instance of `BinaryData`.
+
+In order to deserialize, she simply invokes the `ToObjectFromJson` method:
+
+```C#
+Task RunAsync(BinaryData messageData)
+{
+    var message = messageData.ToObjectFromJson<ResizeImageMessage>();
+
+    foreach (var resolution in message.Resolutions)
+    {
+        var newUrl = GetUrl(resolution.Url, resolution);
+        var image = await DownloadImage(resolution.Url);
+        image.Resize(resolution);
+        await UploadImageAsync(newUrl, image);
+    }
+}
+```
+
+## Requirements
+
+### Goals
+
+* Stable API
+* Make it easy for people to construct bytes from common data types (byte
+  arrays, spans, memories, string, JSON serialized objects)
+* Make it easy for people to convert bytes to common data types of the same
+  representations
+* Provide opinionated defaults in order to reduce complexity
+    - String encoding is assumed to be UTF8
+    - Object serialization is done with `System.Text.Json`, with specific options
+
+### Non-Goals
+
+* Allocation free zero-copying semantics. The primary goal is convenience and
+  usability, not perf. Performance critical scenarios should be enabled by
+  providing overloads with the existing low-level concepts, such as span.
+
+## Design
+
+We're concerned with hard coding the type to `System.Text.Json`, but we
+acknowledge that defaults are very valuable. If we need to support other
+serializers we can provide additional arguments or APIs.
+
+This type can't live in corlib because it would depend on `System.Text.Json`
+(and whatever future serializers it needs to support). Hence, most areas in the
+BCL wouldn't be able to take `BinaryData`, but technologies above the BCL could
+(Microsoft.Extensions, ASP.NET Core, Azure SDK etc). While that might be
+unfortunate, we believe it's acceptable because most of the usability gains
+would be in higher levels anyway.
+
+Thus, we shouldn't think of this as an exchange type in the BCL. We already have
+exchange types; the point of this type is unify them by providing conversions;
+the point isn't to provide yet another type. Thus, we don't see the need to, for
+example, provide new virtual methods on Stream.
+
+We have two options to design the type:
+
+1. We could decide to model the construction via JSON serialization as a factory
+   method. This ensures we can easily support additional serialization formats
+   as necessary and remain linker friendly.
+
+2. But we need to balance implementation flexibility concerns with usability.
+   The primary goal of this type is to provide usability via conveniences. If we
+   don't make the type super usable, the feature will fail. If we feel that the
+   factories aren't discoverable enough, we should combine it with constructors,
+   even if that reduces our flexibility.
+
+### Shipping vehicle
+
+**Assembly**: System.Binary.Data.dll
+**NuGet Package**: System.Binary.Data (stable)
```

Comment:
```suggestion
**Assembly**: System.Memory.Data.dll
**NuGet Package**: System.Memory.Data (stable)
```

**OR**

```suggestion
**Assembly**: System.Data.Binary.dll
**NuGet Package**: System.Data.Binary (stable)
```
I personally prefer the latter as it is more direct in meaning to the API's usage.

Reply:
Oh, that's bug. I mean to write `BinaryDay` because that's the name of the type.